java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java

```diff
@@ -312,5 +315,64 @@ public VectorSchemaRoot slice(int index, int length) {
     return new VectorSchemaRoot(sliceVectors);
   }
 
+  /**
+   * Determine if two record batches are exactly equal.
+   */
+  public boolean equals(VectorSchemaRoot other) {
+    if (other == null) {
+      return false;
+    }
+
+    if (!this.schema.equals(other.schema)) {
+      return false;
+    }
+
+    if (this.rowCount != other.rowCount) {
+      return false;
+    }
+
+    for (int i = 0; i < fieldVectors.size(); i++) {
+      FieldVector vector = fieldVectors.get(i);
+      FieldVector otherVector = other.fieldVectors.get(i);
+      if (!VectorEqualsVisitor.vectorEquals(vector, otherVector)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Determine if two record batches are approximately equal.
```

Comment:
Maybe clarify that this is in regards to floating point values? (I wasn't sure initially if this meant that it was insensitive to schema metadata, for instance, or something else.)

Reply:
I updated the docs, approx equals is in regards to floating point values, other type vectors in VectorSchemaRoot are exactly equals. Both need the same schema metadata.